### DISCUSSION TITLE
Can I use the domain/hostname to connect to the target computer?

### DISCUSSION THREAD

**Discussion by huangsijun17** [rustdesk/rustdesk#5833] [Q&A] [answered]
I hope that I can directly enter the public (sub) domain name or intranet host name of the target computer in RustDesk to connect directly without changing it to an IP address.

**Comment by SergeyMy** [comment, marked as answer]
that's how it works! it should be simple:
`domain_level_x.domain_level_2.domain_level_1:port`
e.g.
`mydomain.com:21118`

**Reply by huangsijun17** [reply]
I didn't enter the :port before, and RustDesk automatically recognized it as an ID.

**Reply by open-trade** [reply]
> hostname:port, e.g. myhost:21118

port is required because RustDesk do not know it is id or hostname

**Reply by DRSDavidSoft** [reply]
> port is required because RustDesk do not know it is id or hostname

I have two problems with this, if the developers are open to discussions:
1. Do RustDesk Ids consists of `.` (dots) in them? If not, you can be sure that a hostname/domain like `mycomputer.lan` or `raspberrypi.local` is in fact NOT an ID, and is a hostname/domain name
2. Even if they collide, you can still do a name resolution at the same time as checking if the entered string is an ID. When RustDesk determines that the entered value is NOT an ID, then if the hostname resolved to an IP address, it could use that to connect.

I believe same thing is happening in TeamViewer/AnyDesk, when you enter a hostname without default port (7070 in AnyDesk or 5938 in TeamViewer), then the user can connect using hostname. Same with Microsoft RDP.

---

I opened this discussion about this problem: https://github.com/rustdesk/rustdesk/discussions/14416
**Update** seems the devs are not open to discussion 😅 closed in 10 mins

**Reply by bluelovers** [reply]
<img width="540" height="170" alt="image" src="https://github.com/user-attachments/assets/757c2023-ab55-404a-b2e0-9a705b0833cc" />

not work

**Reply by huangsijun17** [reply]
> <img alt="image" width="540" height="170" src="https://private-user-images.githubusercontent.com/167966/565284992-757c2023-ab55-404a-b2e0-9a705b0833cc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM4MDQzMTYsIm5iZiI6MTc3MzgwNDAxNiwicGF0aCI6Ii8xNjc5NjYvNTY1Mjg0OTkyLTc1N2MyMDIzLWFiNTUtNDA0YS1iMmUwLTlhNzA1YjA4MzNjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMxOFQwMzIwMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NWQzZDgyZWI0ZjYzNDhkY2Y0NWExZmVjMjcwNTYzYWY3OTc5Y2U1OWIzMmQyMDVjMjcxM2ZkZDZmMGMyYjg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bKVfynBXeq97AODwLXBCKen7i2_nfu6zH6Zp5svTQl0">
> not work

Try ping / telnet it first.

**Comment by huangsijun17** [comment]
> Im am also trying to connect via hostname, but this is not working for me.
> 
> Rustdesk Server is running internally and both clients are also running internally. Server is running on a Linux Ubuntu server, which is not member of the Windows Domain. Client are Windows 11 and part of a Domain. I can connect via ID and also via IP. But not with hostname.
> 
> Tried the hostname only "host123" and also FQDN "host123.xyz.domain.com". But I get only "ID does not exist". Ping to "host123.xyz.domain.com" is also working fine.
> 
> Any suggestions, why connecting to hostname does not work?

try `host123:21118`

**Reply by DRSDavidSoft** [reply]
Now I understand, providing the port (`:21118`) is essential, otherwise RustDesk thinks the hostname is an ID

**Reply by bluelovers** [reply]
<img width="540" height="170" alt="image" src="https://github.com/user-attachments/assets/7be9fe4b-d4ed-4a53-b5e0-5972c12bd238" />

not work

**Comment by FickoDeLux** [comment]
I did a bug report but was almost immediately closed and directed to this topic. So I am not sure what this mean. It is not a bug but a feature or what? Nevertless it is 2026 and still no hostname, hostnam:port not even localhost is working. So if it is a feature than awesome.

**Reply by DRSDavidSoft** [reply]
For me, the `hostname:port` definitely did work correctly, so I believe the developer's intention is to keep this format (instead of hostname-only solution) as a feature.  If using the specified format didn't work, then it is probably a bug, assuming the hostname does indeed correctly resolve.

To test it out, you can do `nslookup myhostname` on Windows, to see if the hostname is resolving correctly.

**Reply by FickoDeLux** [reply]
The hostname is resolving correctly. I have UltraVNC setup parallel to RustDesk and UltaVNC have no problem to connect by name.

**Reply by DRSDavidSoft** [reply]
@FickoDeLux This is likely a misconfiguration, does `ip:port` work when `hostname:port` does not?
